src/convertToObject.js

```diff
@@ -6,7 +6,22 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (sourceString.length === 0) {
+    return {};
+  }
+
+  const stringToObject = sourceString
+    .split(';')
+    .map((value) => {
+      return value.split(':').map((withoutSpace) => withoutSpace.trim());
+    })
+    .reduce((acc, value) => {
+      acc[value[0]] = value[1];
```

Comment:
There is a potential issue here when the split array has a length of 1, which can happen if there's a semicolon without a corresponding style declaration. This would result in trying to access `value[1]` which would be undefined, and thus setting `acc[value[0]]` to undefined. It would be good to check if `value.length` is 2 before setting the property.